MathLib/LinAlg/RowColumnIndices.h

```diff
@@ -14,19 +14,25 @@
 
 namespace MathLib
 {
+// A dummy vector as the default value of the third argument of
+// the constructor.
+static std::vector<std::size_t> dummy_non_ghost_local_ids;
 
 template <typename IDX_TYPE>
 struct RowColumnIndices
 {
 	typedef typename std::vector<IDX_TYPE> LineIndex;
-	RowColumnIndices(LineIndex const& rows_, LineIndex const& columns_)
-		: rows(rows_), columns(columns_)
+	RowColumnIndices(LineIndex const& rows_, LineIndex const& columns_,
```

Comment:
So far, I do not have a more better idea about initialization the reference member. In order make the argument of member of add of class matrix concise, add  ``non_ghost_local_ids`` in this truct is adequate.